tests/unit/backend/corpora/api_server/curator/collection/datasets/test_status.py

```diff
@@ -0,0 +1,34 @@
+from backend.corpora.common.corpora_orm import CollectionVisibility, ProcessingStatus
+from tests.unit.backend.corpora.api_server.base_api_test import BaseAuthAPITest
+
+
+class TestDatasetStatus(BaseAuthAPITest):
+    def test_get_status(self):
+        curator_tag = "tag.h5ad"
+        processing_status = {
+            "processing_status": ProcessingStatus.PENDING.name,
+        }
+        queries = ["dataset_uuid", "curator_tag"]
+        auth_credentials = [
+            (self.make_super_curator_header, "super", 200),
+            (self.get_auth_headers, "owner", 200),
+            (None, "none", 401),
+            (self.make_not_owner_header, "not_owner", 403),
```

Comment:
This should be a 404 (with `404` added to the API `responses` section of the connexion yml as well)

Reply:
Well, actually, I'm unclear on this. I know we decided to use 404 for certain scenarios where the resource exists but the person isn't authorized... and I can't remember what those were...

But since we're allowing collection uuid's to serve as capability entities, _should_ we allow someone to get the processing status of a dataset if they're not the owner but they _do_ have the collection id? 🤔

Reply:
Is there anything a user should not be able to read?

Reply:
changed to 404.